### PR TITLE
fix zopen-clean traversal

### LINKED ISSUE #569: zopen clean --unused cleaning more than it should
In the below command, I specify that I want to clean the bash package, but it is also cleaning other packages containing bash 
```
zopen clean -u bash
/home/itodoro/zopen/usr/local/zopen/bash/bash-5.2.15.20231007_025655.zos
1: bash/bash-5.2.15.20230920_202747.zos <- Removed
2: bash/bash-5.2.15 <- Removed
3: bash/bash-5.2.15.20231007_025655.zos  <-  current
4: cmake/CMake-heads.v3.24.2.20230301_161451.zos/share/bash-completion <- Removed
5: cmake/CMake-heads.v3.27.6/share/bash-completion <- Removed
6: CMake/CMake-heads.v3.27.6/share/bash-completion <- Removed
7: cli/cli-trunk/share/bash-completion <- Removed
8: githubcli/githubcli-DEV/share/bash-completion <- Removed
9: githubcli/githubcli-DEV.20231106_205638.zos/share/bash-completion <- Removed
10: githubcli/githubcli-DEV.20231107_213716.zos/share/bash-completion <- Removed
11: githubcli/githubcli-DEV.20231107_214756.zos/share/bash-completion <- Removed
- Removal of unused package versions complete
```

### PR DESCRIPTION
**Pull request by DevonianTeuchter**
Simplifies logic and should resolve #569 by restricting directory traversal and output commands

### CODE REVIEW

File: bin/zopen-clean [hunk 1]
```diff
@@ -51,60 +51,35 @@ HELPDOC
...
-      current=${i}
-      printInfo "${NC}${GREEN}${i}: ${repo#"${ZOPEN_PKGINSTALL}"/}  <-  current${NC}"
+    printVerbose "Parsing repo: '${repo}' as '${repo#./}'"
+    repo="${repo#./}"
+    if [ "${current}" = "${repo}" ]; then
+      printInfo "${NC}${GREEN}-> ${repo}  <- Current version${NC}"
     else
-      rm -rf "${repo}" > /dev/null 2>&1
-      printInfo "${i}: ${repo#"${ZOPEN_PKGINSTALL}"/} <- Removed"
+      echo rm -rf "${ZOPEN_PKGINSTALL}/${needle}/${repo}" #> /dev/null 2>&1
```
Comment (IgorTodorovskiIBM) [inline bin/zopen-clean:69]: Assuming this `echo` was left over from debugging?

**Comment by DevonianTeuchter** [comment]
> We should add a couple of tests for clean under https://github.com/ZOSOpenTools/metaport/tree/main/tests. We can cover under a new issue, but it seems clean has no coverage as of yet.

Already PR'd:  https://github.com/ZOSOpenTools/metaport/pull/26   :-)

File: bin/zopen-clean (hunk 1 above)
Comment (DevonianTeuchter) [reply bin/zopen-clean]: Yep - just pushed a new commit  [my testing git had become out-of-sync with my committed-git it seems!

**Comment by IgorTodorovskiIBM** [comment]
> > We should add a couple of tests for clean under https://github.com/ZOSOpenTools/metaport/tree/main/tests. We can cover under a new issue, but it seems clean has no coverage as of yet.
> 
> Already PR'd: [ZOSOpenTools/metaport#26](https://github.com/ZOSOpenTools/metaport/pull/26) :-)

Awesome!

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**